### PR TITLE
releaseById changes in order to retrieve the very first release

### PR DESCRIPTION
**Pull request by varnie**
Hello buddy. Please accept and review my pull request. Will be glad to discuss if needed.

### CODE REVIEW

File: src/main/java/com/wavedroid/musicbrainz/api/AlbumResource.java [hunk 1]
```diff
@@ -128,6 +129,10 @@ private Long getReleaseGroupId(Map<String, Object> input) {
         return Long.parseLong(String.valueOf(input.get("release_group_id")));
     }
 
+    private Long getReleaseId(Map<String, Object> input) {
+        return Long.parseLong(String.valueOf(input.get("release_id")));
```
Comment (kagel) [inline src/main/java/com/wavedroid/musicbrainz/api/AlbumResource.java:133]: а нельзя ли тут сразу Long.parseLong(input.get ?
sorry for my english

File: src/main/java/com/wavedroid/musicbrainz/dao/MusicbrainzDao.java [hunk 2]
```diff
@@ -132,7 +132,8 @@
             "         AS tbl) AS tbl2\n" +
             "ORDER BY artist, rg_year, rg_month\n";
 
-    private static final String RELEASE_BY_MBID = "SELECT\n" +
+    // delete, because it is redundant
```
Comment (kagel) [inline src/main/java/com/wavedroid/musicbrainz/dao/MusicbrainzDao.java]: Why not just delete it? We're on git here

File: src/main/java/com/wavedroid/musicbrainz/dao/MusicbrainzDao.java [hunk 3]
```diff
@@ -176,6 +177,61 @@
             "         AS tbl) AS tbl2\n" +
             "ORDER BY artist, rg_year, rg_month\n";
 
+    private static final String RELEASE_BY_MBID = "SELECT\n" +
```
Comment (kagel) [inline src/main/java/com/wavedroid/musicbrainz/dao/MusicbrainzDao.java]: I'd prefer to see a few lines added, not the whole SQL being replaced

File: src/main/java/com/wavedroid/musicbrainz/dao/MusicbrainzDao.java [hunk 4]
```diff
@@ -306,7 +362,7 @@
             "                INNER JOIN release_group r ON c.artist_credit = r.artist_credit\n" +
             "                INNER JOIN release rel ON rel.release_group = r.id\n" +
             "                INNER JOIN medium m ON m.release = rel.id\n" +
-            "              WHERE r.id = ?) AS tbl) AS tbl2\n" +
+            "              WHERE rel.id = ?) AS tbl) AS tbl2\n" +
```
Comment (kagel) [inline src/main/java/com/wavedroid/musicbrainz/dao/MusicbrainzDao.java:318]: А это на что влияло?

**Review by kagel — CHANGES_REQUESTED** [review]

File: src/main/java/com/wavedroid/musicbrainz/api/AlbumResource.java (hunk 1 above)
Comment (varnie) [reply src/main/java/com/wavedroid/musicbrainz/api/AlbumResource.java:133]: можно, изменю.

File: src/main/java/com/wavedroid/musicbrainz/dao/MusicbrainzDao.java (hunk 2 above)
Comment (varnie) [reply src/main/java/com/wavedroid/musicbrainz/dao/MusicbrainzDao.java]: okay

File: src/main/java/com/wavedroid/musicbrainz/dao/MusicbrainzDao.java (hunk 3 above)
Comment (varnie) [reply src/main/java/com/wavedroid/musicbrainz/dao/MusicbrainzDao.java]: okay, I'll provide changes

File: src/main/java/com/wavedroid/musicbrainz/dao/MusicbrainzDao.java (hunk 4 above)
Comment (varnie) [reply src/main/java/com/wavedroid/musicbrainz/dao/MusicbrainzDao.java:318]: по первому альбуму Motorhead возвращалось 13 треков, а не 8.

File: src/main/java/com/wavedroid/musicbrainz/api/AlbumResource.java (hunk 1 above)
Comment (varnie) [reply src/main/java/com/wavedroid/musicbrainz/api/AlbumResource.java:133]: нет, наверное нельзя, не буду изменять (но ради интереса проверю).

Comment (varnie) [reply src/main/java/com/wavedroid/musicbrainz/api/AlbumResource.java:133]: > Error:(134, 40) java: incompatible types: java.lang.Object cannot be converted to java.lang.String

**Comment by varnie** [comment]
feel free to review.

**Comment by varnie** [comment]
Updated according to your comments. Feel free to review, buddy.

**Comment by varnie** [comment]
test

File: src/main/java/com/wavedroid/musicbrainz/dao/MusicbrainzDao.java [hunk 5]
```diff
@@ -163,7 +160,19 @@
             "                re.date_year               AS year,\n" +
             "                re.date_month              AS month,\n" +
             "                m.first_release_date_year  AS rg_year,\n" +
-            "                m.first_release_date_month AS rg_month\n" +
+            "                m.first_release_date_month AS rg_month,\n" +
+            "                row_number() OVER (PARTITION BY r.gid order by\n" +
```
Comment (kagel) [inline src/main/java/com/wavedroid/musicbrainz/dao/MusicbrainzDao.java:164]: изменилось ли время выполнения запроса? :)

**Comment by kagel** [comment]
Hi buddy. Thanks buddy!

**Pull request merged**